modules/web/js/dialog/new-item-dialog.js

```diff
@@ -31,6 +31,10 @@ class NewItemDialog extends ModalDialog {
 
     onSubmit(data, itemName) {
         const app = this._options.application;
+
+        if (!itemName.includes('.bal')) {
```

Comment:
Can use https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith as well. Also, lowercase the itemName and check. Have to consider '.BAL' too.